src/Processors/Formats/Impl/CHColumnToArrowColumn.h

```diff
@@ -34,6 +34,23 @@ class CHColumnToArrowColumn
     };
 
     CHColumnToArrowColumn(const Block & header, const std::string & format_name_, const Settings & settings_);
+    CHColumnToArrowColumn(const ColumnsWithTypeAndName & header_columns_, const std::string & format_name_, const Settings & settings_);
+
+    /// Makes a copy of this converter.
+    /// This can be useful to prepare for conversion in multiple threads.
+    std::unique_ptr<CHColumnToArrowColumn> clone(bool copy_arrow_schema = false) const;
+
+    /// Initializes the arrow schema.
+    /// It's not done in the constructor because LowCardinality column from header always has indexes type UInt8,
+    /// so we should get proper indexes type from the first chunk of data.
+    /// If the arrow schema is already initialized this function does nothing.
+    void initializeArrowSchema(
+        const Chunk * chunk = nullptr,
+        std::optional<size_t> columns_num = std::nullopt,
+        const std::optional<std::unordered_map<String, Int64>> & column_to_field_id = std::nullopt);
+
+    /// Returns the arrow schema (if it's not initialized yet this function will initialize it from the header columns).
+    std::shared_ptr<arrow::Schema> getArrowSchema() const;
```

Comment:
Are we sure that we want to return a shared pointer to a non const member from a const method? I we should either return `std::shared_ptr<const arrow::Schema>` (or `const arrow::Schema &` if we are sure about the lifetime), or make the function non const.

Reply:
Yes, that's what we usually do. However:
1. Functions of the `arrow` library (contrib/arrow) often take `std::shared_ptr<arrow::Schema>` and never `std::shared_ptr<const arrow::Schema>`. See for example [arrow::Table::Make()](https://github.com/ClickHouse/arrow/blob/68dd24ec070e18633a1b78112a463cfa333c6ada/cpp/src/arrow/table.h#L62-L64)
2. All member functions of class [arrow::Schema](https://github.com/ClickHouse/arrow/blob/68dd24ec070e18633a1b78112a463cfa333c6ada/cpp/src/arrow/type.h#L2249) are const, and there are no public member variables. So actually it seems safe to pass `std::shared_ptr<arrow::Schema>` to another thread.

I'll write a comment about that in the code.